### ISSUE TITLE
Wrong url ConnectionString parsing in MongoDB replication setup

### ISSUE THREAD

**Issue by sopanda** [typeorm/typeorm#7130]
## Issue Description

For a connection string with multiple nodes and replicaSet parameter, typeorm v0.2.29 will only use the first found node and remove the replicaSet parameter. The previous version worked correctly because it was just using connection string like it is.

### Expected Behavior

If url is provided by client via connectionString - library will correctly parse and rebuild it with respect to MongoDB replication.

### Actual Behavior

For connectionString in format:
`mongodb://mongodb0.example.com:27017,mongodb1.example.com:27017,mongodb2.example.com:27017/?replicaSet=myRepl`, typeorm v0.2.29 will in the end internally only use first node and parse url to `mongodb://username:password@mongodb0.example.com:27017/` which is used in `mongodb.MongoClient.connect` as a first parameter.

### Steps to Reproduce

The problem is with the next functions inside the library:

1. `DriverUtils.buildDriverOptions()`
2. `MongoDriver.buildConnectionUrl()`

Input:
```typescript
const url = "mongodb://username:password@mongodb0.example.com:27017,mongodb1.example.com:27017,mongodb2.example.com:27017/?replicaSet=myRepl";
const options = DriverUtils.buildDriverOptions({ url });
console.log(JSON.stringify(options, null, 4));
const connectionUrl = buildConnectionUrl(options);
console.log(connectionUrl);
```

Output:
````
{
    "url": "mongodb://username:password@mongodb0.example.com:27017,mongodb1.example.com:27017,mongodb2.example.com:27017/?replicaSet=myRepl",
    "type": "mongodb",
    "host": "mongodb0.example.com",
    "username": "username",
    "password": "password",
    "port": 27017
}
mongodb://username:password@mongodb0.example.com:27017/
````

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |   MacOS  |
| Node.js version     | v14.2.0 |
| Typescript version  | v3.9.7 |
| TypeORM version     | v0.2.29 |

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [x] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [ ] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [x] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by nebkat** [comment]
#7009 related?

**Comment by klassm** [comment]
@nebkat Afaik no. v0.2.30 has the same problem when I tried it out yesterday.

**Comment by imnotjames** [comment]
It kind of looks like it's not even passing that `url` to the upstream mongo driver.

**Comment by imnotjames** [comment]
Fixed by #7908
